iroh-net/src/magicsock.rs

```diff
@@ -742,13 +782,30 @@ impl AsyncUdpSocket for MagicSock {
                             return Poll::Ready(Err(err));
                         }
                         NetworkReadResult::Ok {
-                            bytes,
-                            meta,
+                            mut bytes,
+                            mut meta,
                             source,
                         } => {
-                            buf_out[..bytes.len()].copy_from_slice(&bytes);
+                            let copy_len = if bytes.len() > buf_out.len() {
+                                // split bytes on segment boundary, leaving the largest
+                                // possible number of segments for buf.  Store the remaining
+                                // segements for next round.
+                                let segment_count = buf_out.len() / meta.stride;
```

Comment:
what if `meta.stride` is larger than `buf_out`?

Reply:
that would be a mis-configuration of quinn.  The QUIC MTU is 1200 bytes and quinn should make sure that it calls recv with enough space.

Though you're right that is is a corner case that should be handled.  Because what we should do in that case is return a partial datagram and return the rest of the datagram on the next poll.